nrf_modem/doc/CHANGELOG.rst

```diff
@@ -9,6 +9,61 @@ Changelog
 
 All notable changes to this project are documented in this file.
 
+nrf_modem 2.5.0
+***************
+
+:ref:`Core library <architecture>`
+==================================
```

Comment:
@divipillai Should keep these sections as references to other doc pages? They were intended to indicate what part of the library the change applies to, but they generate links on the right side of the page.
![image](https://github.com/nrfconnect/sdk-nrfxlib/assets/1151289/9a9c47a9-46dd-43af-9b26-a1b8fb331294)
The problem is that when the user clicks those links, they get redirected to another page, instead of navigating in the current page, which is probably what they are trying to do instead (and what happens when they click on the `nrf_modem vX.Y.Z` links instead)

Reply:
We could keep the depth as 1 here to hide the these link on the right side of the page:
![image](https://github.com/nrfconnect/sdk-nrfxlib/assets/91891495/4b02f6b4-b10a-4b34-bc93-704da2847e61)

result:
![image](https://github.com/nrfconnect/sdk-nrfxlib/assets/91891495/f585fe74-4db7-4ef7-9b1c-655f13758a2f)

This could solve this issue. If its okay, I can fix this in the commit I am working on for images?
I would keep these sections as references because user can directly navigate to the respective documents to see the changes.